src/service/video-manager-impl.js

```diff
@@ -376,14 +389,41 @@ class VideoEntry {
    */
   videoPlayed_() {
     this.isPlaying_ = true;
+    this.actionSessionManager_.beginSession();
+    if (this.isVisible_) {
+      this.visibilitySessionManager_.beginSession();
+    }
+    this.analyticsEvent_(VideoAnalyticsEvent.PLAY);
   }
 
   /**
   * Callback for when the video has been paused
    * @private
    */
   videoPaused_() {
+    const previousState = this.getPlayingState();
+    const trackingVideo = assertTrackingVideo_(this.video);
+    if (trackingVideo &&
+        trackingVideo.getCurrentTime() !== trackingVideo.getDuration()) {
+      this.analyticsEvent_(VideoAnalyticsEvent.PAUSE);
+    }
+    this.isPlaying_ = false;
+
+    // Don't trigger session end for autoplay pauses, since session end
+    // is already covered for autoplay in the visibility tracking
+    if (previousState !== PlayingStates.PLAYING_AUTO) {
```

Comment:
hmm, if they set `end-session-when-invisible` to `false` but `include-autoplay` to `true` we do want session to end when autoplay pauses right?